src/Components/Web/src/WebRenderer.cs

```diff
@@ -103,6 +100,44 @@ protected override void Dispose(bool disposing)
         base.Dispose(disposing);
     }
 
+    [UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = "<Pending>")]
+    private void AttachWebRendererInterop(IJSRuntime jsRuntime, JsonSerializerOptions jsonOptions, JSComponentInterop jsComponentInterop)
+    {
+        const string JSMethodIdentifier = "Blazor._internal.attachWebRendererInterop";
+
+        // These arguments should be kept in sync with WebRendererSerializerContext
+        object[] args = [
+            _rendererId,
+            _interopMethodsReference,
+            jsComponentInterop.Configuration.JSComponentParametersByIdentifier,
+            jsComponentInterop.Configuration.JSComponentIdentifiersByInitializer,
+        ];
+
+        if (jsRuntime is IInternalWebJSInProcessRuntime inProcessRuntime)
+        {
+            // Fast path for WebAssembly: Rather than using the JSRuntime to serialize
+            // parameters, we utilize the source-generated WebRendererSerializerContext
+            // for a faster JsonTypeInfo resolution.
+
+            // We resolve a JsonTypeInfo for DotNetObjectReference<WebRendererInteropMethods> from
+            // the JS runtime's JsonConverters. This is because adding DotNetObjectReference<T> as
+            // a supported type in the JsonSerializerContext generates unnecessary code to produce
+            // JsonTypeInfo for all the types referenced by both DotNetObjectReference<T> and its
+            // generic type argument.
+
+            var newJsonOptions = new JsonSerializerOptions(jsonOptions);
```

Comment:
I had another [commit](https://github.com/dotnet/aspnetcore/pull/54956/commits/a10f5c0cbaabf74e91edf059b51c812b2c76b50e) that eliminated this extra `JsonSerializerOptions` and serialized each argument individually, passing a `JsonTypeInfo` for the argument directly and avoiding serializing an `object[]`. There appeared to be a perf benefit when running in the profiler, but it seemed to actually _slightly_ degrade perf on optimized builds (but it was hard to tell), so I reverted it to just go with the simpler approach.